hack/create-custom-cloud-provider-config.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set +o xtrace
+
+# Install kubectl
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+KUBECTL="${REPO_ROOT}/hack/tools/bin/kubectl"
+cd "${REPO_ROOT}" && make "${KUBECTL##*/}"
+
+if [[ -n "${CUSTOM_CLOUD_PROVIDER_CONFIG:-}" ]]; then
+  curl -sL -o tmp_azure_json "${CUSTOM_CLOUD_PROVIDER_CONFIG}"
+  envsubst < tmp_azure_json > azure_json
+  kubectl create secret generic "${CLUSTER_NAME}-control-plane-azure-json" \
```

Comment:
How do we ensure the following:

1) that this pre-created `azure.json` secret won't be overwritten when the mgmt cluster creates a new cluster with the name `${CLUSTER_NAME}` later on in the cluster create flow?
2) that we don't break things if, for example, later on in the flow we attempt to create the "default" azure.json for a particular test flow and receive a "secret already exists" response from the mgmt cluster apiserver

cc @CecileRobertMichon

Reply:
Is it possible that multiple clusters are created in one CI test? At least in our case, we create one workload cluster per test. If we want to create clusters with different configs, we would trigger multiple tests instead of creating multiple clusters in one test.

Reply:
@lzhecheng could correct me since he's been working on the test.

Reply:
The way our azurejson controller is implemented, it will only create a default secret if no secret is already present  https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/111897c28872ac377b6bf572c725d25d10de16a7/controllers/helpers.go#L435

This is meant to allow users to customize cloud provider by pre-creating a secret with their preferred config (including this very use case)

In this case it will see the secret already exists and won't create one. This should work as expected. @nilo19 were you able to verify this works locally?

This is documented here: https://capz.sigs.k8s.io/topics/cloud-provider-config.html

Reply:
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/hack/deploy-cluster-capz.sh#L88
We have a script to deploy a cluster with customizations and it works.